br/pkg/checkpoint/external_storage.go

```diff
@@ -43,9 +79,10 @@ func newExternalCheckpointStorage(
 	ctx context.Context,
 	s storage.ExternalStorage,
 	timer GlobalTimer,
+	flushPosition flushPosition,
```

Comment:
Maybe `flushLocation` or `flushPath`. `position` usually refers to somewhere internal a file if I recall right (consider the `seek` and `open` syscall.)

Reply:
agree, maybe `flushTarget`, `flushDestination`  etc.

Reply:
maybe we can also put `s storage.ExternalStorage` inside the `flushPosition`? feels like they are related

Reply:
done. Actually `flushPath` is embedded in `externalCheckpointStorage`. Maybe we can regard `externalCheckpointStorage` as their same root.
```
type externalCheckpointStorage struct {
	flushPath
	storage storage.ExternalStorage

        // ...
}
```